designdocs/provider_contexts.md

```diff
@@ -0,0 +1,230 @@
+- Is this a set or edit pattern for the applications?
+- Should adding/updating applications be a separate call?
+- Should adding/updating a translation be a separate call?
+
+#### Implementation Suggestion
+
+To limit the amount of unapproved bytes intended to be interpreted as images on chain, an option would be to merely submit the hashes of images for approval.
+This hash would then be allowlisted for that provider.
+Once approved, that provider could submit the image to chain.
+In the governance process, a link to the image would need to be submitted.
+
+### **Example of Wallet Usage via SIWF** <a id='siwf'></a>
+
+SIWF [Signed Request Payload](https://projectlibertylabs.github.io/siwf/v2/docs/DataStructures/All.html) can expand the optional `applicationContext` value with a new optional `id` field that is the text of the `ApplicationIdentifier` on chain.
+This would allow a smooth transition between `applicationContext.url` and `applicationContext.id` with both being optional.
+
+The Wallet would then:
+1. Verify the SIWF Signed Request.
+2. Lookup the Provider via the `publicKey` in the SIWF Signed Request.
+3. If any, fetch the Application Identifier from Frequency.
+4. Display the information from the ApplicationRegistryEntry (or default Provider one) to the user to help them know who they are authorizing.
+5. User is happy to continue with the login process.
+
+### **Provider Dashboard Steps** <a id='dashboard'></a>
+
+Provider Dashboard needs be able to:
+
+- Create a provider without any application context other than the default provider context
+- Update the default provider context, logos, and translations
+- Add new Application Contexts with a unique to that provider id, logos, and translations
+- Remove an existing Application Context
+- Update an Application Contexts with new logos, and translations
+
+
+## Non-goals
+
+- Implementation of verified credentials for applications (Aka FooBar is approved by <Consumer Org>)
+- Providing for independent body to perform verification
+
+## Benefits and Risk
+
+### Benefit: User protection and Application diversity
+
+Users will have a larger trust when they are logging into an application that the application is represented honestly, even with the continuous risk of trusting any new application with user data.
+
+### Risk: Initial structure of only one Frequency Council member approving changes in an application
+
+Tricking a single member is much simpler than a more detailed vetting process; however, at this stage, the number of providers is small.
+
+This process should be re-evaluated as the number of Providers grows.
+
+### Risk: Logo Images on Chain
+
+This provides a direct way to place image content on chain, intended to be interpreted as image content.
+While this must still pass through the governance step to be used by others, there is always a risk of harmful images being proposed.
+This risk is mitigated by:
+- A registered provider must be taking the action
```

Comment:
This is not clear from the sequence diagram, at least for the initial provider creation flow; it looks like all application data (including image bytes) would be recorded on-chain as part of the provider proposal.

Wouldn't the mitigation be submitting the image hashes rather than the images, as discussed in the "Implementation Suggestion" section?

Reply:
Applied some notes to highlight this.